llvm/lib/SYCLPropagateAspectUsage/PropagateAspectUsage.cpp

```diff
@@ -0,0 +1,282 @@
+#include "llvm/SYCLPropagateAspectUsage/PropagateAspectUsage.h"
+#include "../IR/LLVMContextImpl.h"
```

Comment:
I differentiate types by `llvm::Type*` rather than by string names. The only way I found to get pointers for types I was looking for (including DoubleTy) is `NamedStructTypes` in `LLVMContextImpl`. Unfortunately, I didn't find a way to do it in [LLVMContext](https://llvm.org/doxygen/classllvm_1_1LLVMContext.html|https://llvm.org/doxygen/classllvm_1_1LLVMContext.html).

Reply:
For struct types see [my other comment](https://github.com/intel/llvm/pull/5348#discussion_r788820250), for `double` you can just use [`Type::getDoubleTy`](https://llvm.org/doxygen/classllvm_1_1Type.html#acb145f988329d1d621f73abcafea21d8)